Marked busy reply as insecure message

Silverdark opened the pull request:
Currently the busy reply is handled as a normal reply message. As the busy reply is allowed to be sent as insecure message on a secure channel, the following code is getting triggered, which then resets the device:

```csharp
// Prevent plain text message replies when secure channel has been established
if (device.UseSecureChannel && device.IsSecurityEstablished && !reply.IsSecureMessage)
{
    _logger?.LogWarning("An plain text message was received when the secure channel had been established");
    device.CreateNewRandomNumber();
    ResetDevice(device);
    continue;
}
```

Checking the OSDP documentation, the busy reply should be ignored for secure communication:

> In other words, the busy reply is sent outside the secure channel and should not influence the secured messages that are sent before or after this reply.

With this PR busy replies are not longer marked as secure messages. Alternative / better ways to do this are welcome.

bytedreamer commented:
There are a few places where the busy reply type can be checked. Not sure if it makes sense to change the IsSecureMessage logic. It is still a secure message session, however we should allow for clear text response for busy replies. 

The other option is to do the following

```
if (reply.Type != ReplyType.Busy || (device.UseSecureChannel && device.IsSecurityEstablished &&
                                                             !reply.IsSecureMessage))
```

I'll consider which of these options make the most sense. Are you able to verify that it works with a PD that send busy replies while in a Secure Channel session? Thanks

Silverdark commented:
I checked my version again and unfortunately it didn't worked. Looks like I changed it before pushing...

Now I moved the check to the critical region and checked the behavior. 
For me this is working fine with this change. I used a secure channel and I received the busy followed of an ACK reply without any device reset.
